File: Java/src/com/dbc/algorithm/Daily/LongestContinuousIncreasingSubsequence.java
package com.dbc.algorithm.Daily;

public class LongestContinuousIncreasingSubsequence {
    public static int findLengthOfLCIS(int[] nums) {
        if (nums.length == 0)
            return 0;

        int result = 0;
        int count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                count++;
            } else {
                result = Math.max(result, count);
                count = 1;
            }
        }

        return Math.max(result, count);
    }
}

File: src/tmp/Python/array/longest_continuous_increasing_subsequence.py
def findLengthOfLCIS(nums: list) -> int:
    if not nums:
        return 0

    count = 1
    result = 0
    for i in range(1, len(nums)):
        if nums[i] > nums[i - 1]:
            count += 1
        else:
            result = max(result, count)
            count = 1
    
    return max(result, count)

if __name__ == "__main__":
    print(findLengthOfLCIS([]))